fdbserver/BlobManifest.actor.cpp

```diff
@@ -369,7 +401,15 @@ class BlobManifestDumper : public ReferenceCounted<BlobManifestDumper> {
 private:
 	// Read system keys and write to manifest files
 	ACTOR static Future<int64_t> dump(Reference<BlobManifestDumper> self) {
-		state Reference<BlobManifestFileSplitter> splitter = self->fileSplitter_;
+		state Arena arena;
+		DatabaseConfiguration config = wait(getDatabaseConfiguration(self->db_, true));
+		if (config.encryptionAtRestMode.isEncryptionEnabled()) {
```

Comment:
Nit (mild optimization): the EAR mode cannot change at runtime once the cluster is created, so we technically only need to read the db config once at startup instead of every manifest dump

Reply:
Also related to testing - we should add manifest dumping to BlobGranuleApiCorrectness and add code probes in encryptionAtRestPlaintextMarkerCheck to ensure that the manifest files are checked as part of that workload

Reply:
+1 - currently we have simulation tests to scan on-disk files for:
1. TLogs
2. Redwood files
3. KVBackup files
4. BlobGranules

I think adding BlobManifest to the mix should extend the coverage for Backup/Restore as well (?)

Reply:
currently manifest files are created under file:///simfdb/fdbblob/manifest, so it's already covered by BlobGranules?..

Reply:
Interesting, then it would be good to check why isn't the test failing without your fix?

Reply:
One possibility is: manifest file never contains `full key` and only stores the partial keys all the time? For KVBackup/Restore we found the encryption `gap` running this test, as for optimization Manifest was storing full keys there in some scenarios.

Reply:
I think I got the cause. I didn't inject the marker into manifest file, so the check always succeed. 

Fixed now. With this PR, if manifest file is not encrypted, BlobGranuleApiCorrectness fails.